src/main/java/com/google/sps/servlets/AddTA.java

```diff
@@ -30,30 +35,39 @@ public void init(ServletConfig config) throws ServletException {
     }
   }
 
-  // Add a TA to the datastore
+  // Add a TA to the class datastore
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
+    
     try {
       // Obtain the teaching assistant email and search for the user
       String teachingAssistantEmail = request.getParameter("taEmail").trim();
       UserRecord userRecord = authInstance.getUserByEmail(teachingAssistantEmail);
 
+      // Find the corresponding Class Entity
       String classCode = request.getParameter("classCode").trim();
+      Key classKey = KeyFactory.stringToKey(classCode);
+      Entity classEntity = datastore.get(classKey);
 
-      // Create a TA entity with both a user ID and a class ID
-      Entity taEntity = new Entity("TA");
-      taEntity.setProperty("userKey", userRecord.getUid());
-      taEntity.setProperty("classKey", classCode);
+      // Get the list of TAs
+      ArrayList<String> listOfClassTAs = (ArrayList) classEntity.getProperty("taList");
```

Comment:
This should be done in a `Transaction` that way the list won't get messed up if two people add ta's at the same time.